### PR TITLE
Adding solution for task PianoKeys

### PR DESCRIPTION
**Pull request by sarunaspra**
Not sure if I fully fullfiled task requirements, because, for example, in my solution case Key(Note.C, Accidental.Sharp, Octave.First) and Key(Note.D, Accidental.Flat, Octave.First) are not equal, but according to the task description they should be equal.

### CODE REVIEW

File: Homework3/PianoKeys/Structures/PianoKey.cs [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+            Accidental = accidental;
+        }
+        #endregion
+
+        #region Public Methods
+        public int CompareTo(PianoKey other)
+        {
+            int octaveComparisonResult = Octave.CompareTo(other.Octave);
+            int noteComparisonResult = Note.CompareTo(other.Note);
+            int AccidentalComparisionResult = Accidental.CompareTo(other.Accidental);
```
Comment (sarunaspra) [inline Homework3/PianoKeys/Structures/PianoKey.cs:64]: Sorry, my mistake with the variable naming, I'll fix it.

File: Homework3/PianoKeys/Structures/PianoKey.cs [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        public PianoKey(Octave octave, Note note, Accidental accidental)
+        {
+            Octave = octave;
+            Note = note;
+            Accidental = accidental;
+        }
+        #endregion
+
+        #region Public Methods
+        public int CompareTo(PianoKey other)
```
Comment (radislavkireev) [inline Homework3/PianoKeys/Structures/PianoKey.cs:60]: Check only Octave - pls see requirements https://drive.google.com/drive/folders/1lbNAaiMZsgLovlXhq2n9z_f-arH8pQ4Y 

```
// C# of the first octave
Key c = new Key(Note.C, Accidental.Sharp, Octave.First);
Console.WriteLine(c); // C# (1)
// D flat of the first octave
Key d = new Key(Note.D, Accidental.Flat, Octave.First);
Console.WriteLine(c.Equals(d)); // True, is the same key!
```

File: Homework3/PianoKeys/Structures/PianoKey.cs [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+                    }
+                }
+            }
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (!(obj is PianoKey))
+            {
+                throw new ArgumentException("Cannot compare to given object type");
```
Comment (radislavkireev) [inline Homework3/PianoKeys/Structures/PianoKey.cs:106]: just return false instead

File: Homework3/PianoKeys/Structures/PianoKey.cs [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+                    {
+                        return 0;
+                    }
+                }
+            }
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (!(obj is PianoKey))
```
Comment (radislavkireev) [inline Homework3/PianoKeys/Structures/PianoKey.cs:104]: Add the case when obj is null

**Review by radislavkireev — CHANGES_REQUESTED** [review]

File: Homework3/PianoKeys/Structures/PianoKey.cs (hunk 2 above)
Comment (sarunaspra) [reply Homework3/PianoKeys/Structures/PianoKey.cs:60]: So if Octaves of Keys coincide, the result is that Keys are equal?

Comment (radislavkireev) [reply Homework3/PianoKeys/Structures/PianoKey.cs:60]: Lets keep it as is, I have no certain understanding in the domain, how to compare pianos?

**Review by radislavkireev — APPROVED** [review]

**Pull request merged**